app/views/setup/welcome.html.slim

```diff
@@ -8,6 +8,9 @@
 h1 Initial CaaSP Configuration
 = form_for :settings, url: setup_path, method: :put do |f|
   .form-group
+    = f.label :dashboard, "Dashboard location"
+    = f.text_field :dashboard, value: (Pillar.value(pillar: :dashboard) || request.host), class: "form-control"
```

Comment:
Do you explicitly check that the `dashboard_ref != "localhost"`? As you said, that would always fail...